### PR TITLE
Adds Different Initiliazers for SourceLocation and SourceRange.

### PR DESCRIPTION
**Pull request by achrafmam2**
Adds getFile(for:)

### CODE REVIEW

File: Sources/Clang/Token.swift [hunk 1]
```diff
@@ -79,7 +79,36 @@ func convertToken(_ clang: CXToken) -> Token {
...
+  }
+
+  /// Creates a source location associated with a given file/line/column
+  /// in a particular translation unit
+  /// - parameters:
+  ///   - tu: The translation unit associated with the location to extract.
+  ///   - file: Source file.
+  ///   - line: The line number in the source file.
+  ///   - column: The column number in the source file.
+  init(tu: TranslationUnit, file: File, line: Int, column: Int) {
```
Comment (CodaFi) [inline Sources/Clang/Token.swift]: Nit: Spell `translationUnit` out or make this parameter have no label `_`.

File: Sources/Clang/Token.swift [hunk 2]
```diff
@@ -79,7 +79,36 @@ func convertToken(_ clang: CXToken) -> Token {
...
+      clang_getLocation(tu.clang, file.clang, UInt32(line), UInt32(column))
+  }
+
+  /// Creates a source location associated with a given character offset
+  /// in a particular translation unit
+  /// - parameters:
+  ///   - tu: The translation unit associated with the location to extract.
+  ///   - file: Source file.
+  ///   - offset: character offset in the source file.
+  init(tu: TranslationUnit, file: File, offset: Int) {
```
Comment (CodaFi) [inline Sources/Clang/Token.swift]: Same here

File: Sources/Clang/Token.swift (hunk 1 above)
Comment (achrafmam2) [reply Sources/Clang/Token.swift]: Done!

File: Sources/Clang/Token.swift (hunk 2 above)
Comment (achrafmam2) [reply Sources/Clang/Token.swift]: Done!

**Comment by CodaFi** [comment]
⛵️

**Comment by harlanhaskins** [comment]
@achrafmam2 are you in the LLVMSwift Slack?

llvmswift-slack.herokuapp.com

**Comment by achrafmam2** [comment]
No. I am not.

**Pull request merged**